app-vite/types/configuration/framework-conf.d.ts

```diff
@@ -13,40 +15,91 @@ type SerializableConfiguration<T> = {
 };
 
 interface QuasarFrameworkConfiguration {
+  /**
+   * @see - QuasarConfOptions tab in API cards throughout the docs
+   */
   config?: SerializableConfiguration<QuasarUIConfiguration>;
+  /**
+   * One of the Quasar IconSets
+   *
+   * @see https://v2.quasar.dev/options/quasar-icon-sets
+   *
+   * @example 'material-icons'
+   */
   iconSet?: QuasarIconSets;
+  /**
+   * One of the Quasar language packs
+   *
+   * @see https://v2.quasar.dev/options/quasar-language-packs
+   *
+   * @example 'en-US'
+   * @example 'es'
+   */
   lang?: QuasarLanguageCodes;
+  /**
+   * Quasar CSS addons have breakpoint aware versions of flex and spacing classes
+   *
+   * @see https://quasar.dev/layout/grid/introduction-to-flexbox#flex-addons
+   * @see https://quasar.dev/style/spacing#flex-addons
+   */
   cssAddon?: boolean;
 
   /**
    * Auto import - how to detect components in your vue files
    *   "kebab": q-carousel q-page
    *   "pascal": QCarousel QPage
    *   "combined": q-carousel QPage
+   *
    * @default 'kebab'
    */
   autoImportComponentCase?: "kebab" | "pascal" | "combined";
-
   /**
    * Auto import - which file extensions should be interpreted as referring to Vue SFC?
-   * @default [ 'vue' ]
+   *
+   * @default ['vue']
    */
   autoImportVueExtensions?: string[];
-
   /**
    * Auto import - which file extensions should be interpreted as referring to script files?
-   * @default [ 'js', 'jsx', 'ts', 'tsx' ]
+   *
+   * @default ['js', 'jsx', 'ts', 'tsx']
    */
   autoImportScriptExtensions?: string[];
-
   /**
    * Treeshake Quasar's UI on dev too?
    * Recommended to leave this as false for performance reasons.
+   *
    * @default false
    */
   devTreeshaking?: boolean;
 
-  components?: (keyof QuasarPluginOptions["components"])[];
-  directives?: (keyof QuasarPluginOptions["directives"])[];
-  plugins?: (keyof QuasarPluginOptions["plugins"])[];
+  /**
+   * Quasar will auto import components based on your usage.
+   * But, in case you have a special case, you can manually specify Quasar components to be available everywhere.
+   *
+   * An example case would be having custom component definitions with plain string templates, inside .js or .ts files,
```

Comment:
> An example case would be having custom component definitions with plain string templates, inside .js or .ts files,

An example case would be to have custom component definitions with plain string templates inside .js or .ts files,